addons/portal/models/res_partner.py

```diff
@@ -1,12 +1,34 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import fields, models
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
+    is_default_billing_address = fields.Boolean(default=False, store=True)
+    is_default_shipping_address = fields.Boolean(default=False, store=True)
+
+    def _update_delivery_and_shipping_address(self, partner_id, address_type, **kw):
+        partner_sudo = self.browse(partner_id)
```

Comment:
:thinking: I guess but why not just use self and do ensure_one since we seem to only call this with partner_sudo in the first place